cachito/common/checksum.py

```diff
@@ -0,0 +1,39 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import hashlib
+from typing import Optional
+from cachito.errors import UnknownHashAlgorithm
+
+
+def get_hasher(algorithm: Optional[str] = None):
+    """Get a hasher by a specific algorithm.
+
+    :param str algorithm: the algorithm name used to get a hasher from the hashlib module.
+    :return: the hasher got from the hashlib module.
+    :raise UnknownHashAlgorithm: if the algorithm cannot be found.
+    """
+    try:
+        return hashlib.new(algorithm or "sha256")
+    except ValueError:
+        raise UnknownHashAlgorithm(f"Hash algorithm {algorithm} is unknown.")
+
+
+def compute_file_checksum(hasher, file_path: str, chunk_size: Optional[int] = None) -> str:
+    """Compute checksum of a file.
+
+    :param hasher: a hasher got from the hashlib module. Use :func:`get_hasher` to get a hasher,
+        or just call hashlib method to get one, e.g. ``hashlib.sha256``.
+    :param str file_path: compute checksum for this file.
+    :param chunk_size: the optional chunk size passed to fileobject.read method.
+    :type chunk_size: int or None
+    :return: the computed checksum.
+    :rtype: str
+    """
+    _chunk_size = 10240 if chunk_size is None else chunk_size
+    with open(file_path, "rb") as f:
+        while True:
+            chunk = f.read(_chunk_size)
+            if not chunk:
+                break
```

Comment:
```suggestion
        while chunk := f.read(_chunk_size):
```
Optional, just a nice use case for the walrus operator :D

Reply:
Cool!